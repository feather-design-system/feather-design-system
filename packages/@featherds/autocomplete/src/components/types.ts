import { Ref, PropType } from "vue";
import HighlightProps from "./Highlight/HighlightProps";
import { InputWrapperProps, InputSubTextProps } from "@featherds/input-helper";
interface IAutocompleteItem extends IAutocompleteItemType {
  _new?: string;
  _text: string;
  _pre?: IAutocompleteChipIcon;
}
interface IAutocompleteItemType {
  [k: string]: unknown;
}

interface IAutocompleteGridColumn {
  prop: keyof IAutocompleteItemType;
  title: string;
  align?: "right" | "left" | "center";
}

interface IAutocompleteResultRender {
  reset: () => void;
  handleKeyPress: (
    e: KeyboardEvent,
    internalResults: IAutocompleteItemType[]
  ) => boolean;
  active: {
    row: number;
    col?: number;
  };
  first: () => void;
}

export enum AutocompleteTypes {
  multi = "multi",
  single = "single",
}
interface IStrategyOptions {
  selectionLimit: Ref<number>;
  selectionLimitReached: Ref<boolean>;
  modelValue: Ref<IAutocompleteItemType | IAutocompleteItemType[]>;
  textProp: Ref<string>;
  input: Ref<HTMLInputElement>;
  query: Ref<string>;
  emitSearch: () => void;
  forceCloseResults: Ref<boolean>;
  allowNew: Ref<boolean>;
  internalResults: Ref<IAutocompleteItemType[]>;
}

interface IAutocompleteChipIcon {
  icon: unknown;
  title: string;
}

export type {
  IStrategyOptions,
  IAutocompleteGridColumn,
  IAutocompleteItemType,
  IAutocompleteItem,
  IAutocompleteChipIcon,
  IAutocompleteResultRender,
};

export const LABELS = {
  noResults: "No results found",
  minChar: "Enter ${min} characters to search",
  clear: "Clear value",
  selectionLimit: "Selection limit reached",
  new: "New",
  remove: "Remove",
};
export const props = {
  id: {
    type: String,
  },
  modelValue: {
    type: [Array, Object] as PropType<
      IAutocompleteItemType[] | IAutocompleteItemType
    >,
  },
  results: {
    type: Array as PropType<IAutocompleteItemType[]>,
    default: () => [],
  },
  textProp: {
    type: String as unknown as PropType<keyof IAutocompleteItemType>,
    default: "_text",
  },
  loading: {
    type: Boolean,
    default: false,
  },
  minChar: {
    type: Number,
    default: 0,
  },
  allowNew: {
    type: Boolean,
    default: false,
  },
  selectionLimit: {
    type: Number,
  },
  newMatcher: {
    type: Function,
    default: (
      item: IAutocompleteItemType,
      query: string,
      comp: { textProp: string }
    ) => {
      return (
        (item[comp.textProp] as string).toString().toLowerCase() ===
        query.toLowerCase()
      );
    },
  },
  type: {
    type: String as PropType<keyof typeof AutocompleteTypes>,
    required: true,
    validator: (v: keyof typeof AutocompleteTypes) => {
      // The value must match either
      return !!AutocompleteTypes[v];
    },
  },
  labels: {
    type: Object as PropType<Partial<typeof LABELS>>,
    default() {
      return LABELS;
    },
  },
  gridConfig: {
    type: Array as PropType<IAutocompleteGridColumn[]>,
  },
  schema: {
    type: Object,
    required: false,
  },
  ...HighlightProps,
  ...InputSubTextProps,
  ...InputWrapperProps,
} as const;

export const emits = {
  "update:modelValue": (
    value: IAutocompleteItemType | IAutocompleteItemType[] | undefined
  ) => true,
  new: (value: string) => true,
  search: (value: string) => true,
};
