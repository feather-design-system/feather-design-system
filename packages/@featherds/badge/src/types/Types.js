const TYPES = {
  INFO: "info",
  ERROR: "error",
};

export default TYPES;
