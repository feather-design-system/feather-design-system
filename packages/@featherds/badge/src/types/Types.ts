export enum BadgeTypes {
  info = "info",
  error = "error",
}

export interface IBadge {
  type: BadgeTypes;
  title: string;
}
