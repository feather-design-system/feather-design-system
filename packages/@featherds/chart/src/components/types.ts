export type ShirtSize = "xs" | "sm" | "md" | "lg" | "xl";

export type FeatherChartType = "bar";

export type FeatherBarChartData = {
  labels: Array<string>;
  datasets: Array<{
    label: string;
    data: Array<string | number>;
    backgroundColor?: string;
  }>;
};

export type FeatherBarChartOptions = {
  responsive?: boolean;
  maintainAspectRation?: boolean;
};

export type FeatherD3Dimensions = {
  height: number;
  width: number;
  margin: {
    top: number;
    right: number;
    bottom: number;
    left: number;
  }
};

export type FeatherD3Options = {
  units: string;
  class?: "color1" | "color2";
};


// Data can be any array of objects returned from an API or local JSON
export type FeatherD3Data = Array<unknown>;

export interface ICoordinateAxes {
  x: string,
  y: string
}
