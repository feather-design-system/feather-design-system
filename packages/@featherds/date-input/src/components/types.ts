export interface IDateDisabledConfig extends IDateRange {
  dates?: (Date | number)[];
  ranges?: IDateRange[];
  days?: number[];
  daysOfMonth?: number[];
  customPredictor?: (date: Date | number) => boolean;
}
interface IDateRange {
  to?: Date | number;
  from?: Date | number;
}
export const LABELS = {
  day: "Day",
  month: "Month",
  year: "Year",
  prevMonth: "Previous month",
  nextMonth: "Next month",
  selectMonth: "Select month",
  selectYear: "Select year",
  inCalendar: "In calendar",
  calendar: "Click for calendar",
  clear: "Clear date",
  menu: "Use arrow keys to navigate dates. Page down and page up will navigate by month. Shift page down and shift page up will navigate by year. Press escape to exit the calendar.",
};
