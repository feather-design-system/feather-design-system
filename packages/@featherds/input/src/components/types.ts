import { InputWrapperProps, InputSubTextProps } from "@featherds/input-helper";
import { PropType } from "vue";
export const props = {
  type: {
    type: String,
    default: "text",
  },
  modelValue: {
    type: [String, Number] as PropType<string | number>,
  },
  maxlength: {
    type: Number,
    required: false,
    default: 0,
  },
  schema: {
    type: Object as PropType<Record<string, any>>,
    required: false,
  },
  id: {
    type: String,
    required: false,
  },
  ...InputWrapperProps,
  ...InputSubTextProps,
} as const;

export const emits = {
  "update:modelValue": (value: string | undefined | number) => true,
};
