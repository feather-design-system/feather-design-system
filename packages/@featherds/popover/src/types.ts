// Note: introduced new types
export type Tooltip = {
  height: number;
  width: number;
};

export type Trigger = {
  height: number;
  width: number;
  left: number;
  top: number;
};

export enum PopoverPlacement {
  top = "top",
  bottom = "bottom",
  left = "left",
  right = "right",
}

export enum PointerAlignment {
  center = "center",
  left = "left",
  right = "right",
}
