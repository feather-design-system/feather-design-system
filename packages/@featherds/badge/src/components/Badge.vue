<template>
  <div class="badge" :class="[type, siblings ? 'siblings' : '']">
    <div class="badge-icon" :title="title"></div>
  </div>
</template>
<script>
export default {
  props: {
    type: { type: String, required: true },
    title: { type: String, required: true },
    siblings: { type: Boolean, default: false },
  },
};
</script>
<style lang="scss" scoped>
@import "~@featherds/styles/themes/variables";
.badge {
  width: 10px;
  height: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  flex: none;
  &.info {
    .badge-icon {
      border-width: 5px;
      border-style: solid;
      border-color: var($primary);
      border-radius: 100%;
      height: 10px;
      width: 10px;
    }
    &.siblings {
      margin-right: -2px;
      .badge-icon {
        border-right-color: transparent;
      }
    }
  }
  &.error .badge-icon {
    background-color: var($error);
    transform: rotate(45deg);
    transform-origin: center center;
    width: 7px;
    height: 7px;
  }
}
</style>
