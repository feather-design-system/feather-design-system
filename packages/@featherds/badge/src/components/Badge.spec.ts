import Badge from "./Badge.vue";
import { BadgeTypes } from "../types/Types";
import { shallowMount } from "@vue/test-utils";

const getWrapper = function (options: Record<string, unknown>) {
  return shallowMount(Badge, options);
};

describe("Badge.vue", () => {
  it("should render correct type with title", async () => {
    const wrapper = getWrapper({
      props: {
        type: BadgeTypes.error,
        title: "TEST",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    await wrapper.setProps({
      type: BadgeTypes.info,
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render siblings modifier", async () => {
    const wrapper = getWrapper({
      props: {
        type: BadgeTypes.error,
        title: "TEST",
        siblings: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
