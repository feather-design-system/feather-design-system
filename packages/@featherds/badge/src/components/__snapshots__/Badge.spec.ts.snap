// Vitest Snapshot v1

exports[`Badge.vue > should render correct type with title 1`] = `
<div
  class="badge error"
>
  <div
    class="badge-icon"
    title="TEST"
  />
</div>
`;

exports[`Badge.vue > should render correct type with title 2`] = `
<div
  class="badge info"
>
  <div
    class="badge-icon"
    title="TEST"
  />
</div>
`;

exports[`Badge.vue > should render siblings modifier 1`] = `
<div
  class="badge error siblings"
>
  <div
    class="badge-icon"
    title="TEST"
  />
</div>
`;
